src/Elliptic/Actions/InitializeSystem.hpp

```diff
@@ -0,0 +1,125 @@
+ * \note Currently the sources are always retrieved from an analytic solution.
+ *
+ * Uses:
+ * - Metavariables:
+ *   - `analytic_solution_tag`
+ * - System:
+ *   - `volume_dim`
+ *   - `fields_tag`
+ *   - `gradient_tags`
+ * - DataBox:
+ *   - `Tags::Mesh<volume_dim>`
+ *   - `Tags::Coordinates<Dim, Frame::Inertial>`
+ *   - All items required by the added compute tags
+ *
+ * DataBox:
+ * - Adds:
+ *   - `fields_tag`
+ *   - `db::add_tag_prefix<LinearSolver::Tags::OperatorAppliedTo, fields_tag>`
+ *   - `db::add_tag_prefix<Tags::Source, fields_tag>`
+ *   - `db::add_tag_prefix<LinearSolver::Tags::Operand, fields_tag>`
+ *   - `db::add_tag_prefix<LinearSolver::Tags::OperatorAppliedTo,
+ *   db::add_tag_prefix<LinearSolver::Tags::Operand, fields_tag>>`
+ *   - `Tags::deriv<variables_tag, volume_dim, Frame::Inertial>>`
+ */
+struct InitializeSystem {
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            size_t Dim, typename ActionList, typename ParallelComponent>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ElementIndex<Dim>& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using system = typename Metavariables::system;
+    using fields_tag = typename system::fields_tag;
+    using linear_operator_applied_to_fields_tag =
+        db::add_tag_prefix<LinearSolver::Tags::OperatorAppliedTo, fields_tag>;
+    using sources_tag = db::add_tag_prefix<::Tags::Source, fields_tag>;
+    using linear_operand_tag =
+        db::add_tag_prefix<LinearSolver::Tags::Operand, fields_tag>;
+    using linear_operator_applied_to_operand_tag =
+        db::add_tag_prefix<LinearSolver::Tags::OperatorAppliedTo,
+                           linear_operand_tag>;
+    using inv_jacobian_tag =
+        ::Tags::InverseJacobian<::Tags::ElementMap<Dim>,
+                                ::Tags::Coordinates<Dim, Frame::Logical>>;
+
+    using simple_tags =
+        db::AddSimpleTags<fields_tag, linear_operator_applied_to_fields_tag,
+                          sources_tag, linear_operand_tag,
+                          linear_operator_applied_to_operand_tag>;
+    using compute_tags = db::AddComputeTags<
+        // The gradients are needed by the elliptic operator
+        ::Tags::DerivCompute<linear_operand_tag, inv_jacobian_tag,
+                             typename system::gradient_tags>>;
+
+    const auto& mesh = db::get<::Tags::Mesh<Dim>>(box);
+    const size_t num_grid_points = mesh.number_of_grid_points();
+    const auto& inertial_coords =
+        get<::Tags::Coordinates<Dim, Frame::Inertial>>(box);
+
+    // Set initial data to zero. Non-zero initial data would require us to also
+    // compute the linear operator applied to the the initial data.
```

Comment:
Please comment why you can't set this to signaling nan instead

Reply:
see comment above please